.chloggen/ottl-dynamic-keep-keys.yaml

```
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. receiver/filelog)
+component: pkg/ottl
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Support dynamic keys in `keep_keys` and `keep_matching_keys` functions, allowing the keys to be specified at runtime.
```

```suggestion
note: Support paths and expressions as keys in `keep_keys` and `keep_matching_keys`
```
I think this makes it a bit clearer.